### PR TITLE
Split DB_POOL_SIZE & DB_MIN_IDLE environment variables

### PR DESCRIPTION
**Pull request by justahero**
This PR splits the existing environment variables `DB_POOL_SIZE` & `DB_MIN_IDLE` into separate variables to provide different connection settings to the primary & replica databases. The main reason for this change is to be able to configure different values for both database pools.

The new env variables are:

* `DB_PRIMARY_POOL_SIZE` & `DB_REPLICA_POOL_SIZE` to configure the number of connections in the primary & replica pool
* `DB_PRIMARY_MIN_IDLE` & `DB_REPLICA_MIN_IDLE` to set the number of idle connections each pool at least maintains
* update descriptions of env vars.

**Note** the descriptions of the `*_POOL_SIZE` variables in `app.json` are most likely off & need correct wording.

### CODE REVIEW

File: src/config/database_pools.rs [hunk 1]
```diff
@@ -20,15 +24,30 @@ pub struct DatabasePools {
...
+    pub fn primary_max_pool_size(&self) -> u32 {
+        self.primary.pool_size
+    }
+
+    pub fn replica_max_pool_size(&self) -> u32 {
+        self.replica
+            .as_ref()
+            .map(|config| config.pool_size)
+            .unwrap_or(Self::DEFAULT_POOL_SIZE)
+    }
```
Comment (pietroalbini) [inline src/config/database_pools.rs:45]: This function doesn't seem to be used anywhere.

File: src/config/database_pools.rs [hunk 2]
```diff
@@ -20,15 +24,30 @@ pub struct DatabasePools {
...
 }
 
 impl DatabasePools {
     pub fn are_all_read_only(&self) -> bool {
         self.primary.read_only_mode
     }
+
+    pub fn primary_max_pool_size(&self) -> u32 {
+        self.primary.pool_size
+    }
```
Comment (pietroalbini) [inline src/config/database_pools.rs:38]: I'm not sure if it's worth having a function just for this, the rest of the code could access the pool size directly.

File: src/config/database_pools.rs (hunk 1 above)
Comment (justahero) [reply src/config/database_pools.rs]: Good catch!

File: src/config/database_pools.rs (hunk 2 above)
Comment (justahero) [reply src/config/database_pools.rs]: Removed this to be consistent.

**Comment by pietroalbini** [comment]
Thanks!

@bors r+

**Pull request merged**